addons/mrp_subcontracting_dropshipping/models/stock_picking.py

```diff
@@ -11,3 +11,14 @@ def _get_warehouse(self, subcontract_move):
         if subcontract_move.sale_line_id:
             return subcontract_move.sale_line_id.order_id.warehouse_id
         return super(StockPicking, self)._get_warehouse(subcontract_move)
+
+    def _action_done(self):
+        res = super(StockPicking, self)._action_done()
+        finished_moves = self.env['stock.move']
+        for picking in self:
+            done_production = picking._get_subcontract_production().filtered(lambda p: p.state == 'done')
+            for move in done_production.move_finished_ids:
+                finished_moves |= move
+        if finished_moves:
```

Comment:
Does _trigger_assign() is not better in this case than looping on move_dest_ids ?